### ISSUE TITLE
Create an example for different usage of statements : defer, panic and recover

### ISSUE THREAD

**Issue by Nikoltod** [Nikoltod/Hello-Go#1]
You can use this article from the [Go Blog](https://blog.golang.org/defer-panic-and-recover)
Here's an example which defer statement will execute first - a basic understanding of how to "defer" is used.

```
package main

import 'fmt'

func main() {
	//The first defer will be executed last.
	defer fmt.Println("1")

	//The last defer will be executed first.
	defer fmt.Println("2")

	//This func call will be executed first of all.
	fmt.Println("3")
}
```

Another example for "defer" statement can be the following 

```
package main

import "fmt"

//1. A deferred function's arguments are evaluated when the defer statement is evaluated.
func printZero() {
      i := 0
      defer fmt.Println(i)
      // i will be '0' because the value of 'i' will be incremented after the execution of fmt.Println(i) will occur
      i++
      return
}

//2. Deferred function calls are executed in Last In First Out order after the surrounding function returns.
func execCount() {
      //This loop will print 43210 instead of 01234 - because the "defer" execution is LIFO - Last In First Out
       for i:=0; i < 5; i++ {
            defer fmt.Println(i)
       }
}

//This can be quite dangerous - meaning that values can be changed in compile time.
//3. Deferred functions may read and assign to the returning function's named return values.
func reasignReturn() {
        defer func() { i++ }()
        return 1
}

func main() {
     //I will call the functions in sequence 3,2,1
    defer printZero()
    defer execCount()
    defer reasignReturn()
}
```
The call count from "main()" will be  1. reasingReturn() 2. execCount() 3. printZero()

"defer" can be used to handle errors and func calls for a nice and controlled execution.

Here's an good example for "panic()" and "recovery()"

```
package main

import "fmt"

func a(i int) {
      if i > 3 {
           fmt.Println("Panic !")
           panic(fmt.Sprintf("%v", i))
      }
      defer fmt.Println("Defer from a", i)
      fmt.Println(State of a", i)
     a(i + 1)
}

func b() {
      defer func() {
            if r := recover(); r != nil {
               fmt.Println("Recover in b", r)
      }()
      fmt.Println("Calling a")
      a(0)
      fmt.Println("Recover from b")
}

func main() {
      b()
      fmt.Println("Rerutned from b")
}
```
The output will be :

```
Calling a
State of a 0
State of a 1
State of a 2
State of a 3
Panic!
Defer in a 3
Defer in a 2
Defer in a 1
Defer in a 0
Recover from b 4
Returned normally from b
```

**Panic** is a built-in function that stops the ordinary flow of control and begins panicking. When the function F calls panic, execution of F stops, any deferred functions in F are executed normally, and then F returns to its caller. To the caller, F then behaves like a call to panic. The process continues up the stack until all functions in the current goroutine have returned, at which point the program crashes. Panics can be initiated by invoking panic directly. They can also be caused by runtime errors, such as out-of-bounds array accesses.

**Recover** is a built-in function that regains control of a panicking goroutine. Recover is only useful inside deferred functions. During normal execution, a call to recover will return nil and have no other effect. If the current goroutine is panicking, a call to recover will capture the value given to panic and resume normal execution.

In summary, the defer statement (with or without panic and recover) provides an unusual and powerful mechanism for control flow. It can be used to model a number of features implemented by special-purpose structures in other programming languages.
